app/services/forest_liana/two_factor_registration_confirmer.rb

```diff
@@ -0,0 +1,36 @@
+module ForestLiana
+  class TwoFactorRegistrationConfirmer
+    def initialize(
+      project_id,
+      use_google_authentication,
+      auth_data
+    )
+      @project_id = project_id
+      @use_google_authentication = use_google_authentication
+      @auth_data = auth_data
+    end
+
+    def perform
+      begin
+        body_data = { 'useGoogleAuthentication' => @use_google_authentication }
+
+        if @use_google_authentication
+          body_data['forestToken'] = @auth_data[:forest_token]
+        else
+          body_data['email'] = @auth_data[:email]
+        end
+
+        response = ForestApiRequester.post(
+          "/liana/v2/projects/#{@project_id}/two-factor-registration-confirm",
+          body: body_data,
+        )
```

Comment:
If for this call, the server responds 404 or 422, the `ForestApiRequest.check_configuration_errors` will display an error message that is not really appropriate as we don't use `envSecret`, `renderingId` here.

Reply:
It looks like you've gone too far by factorising the basic errors in `ForestApiRequest`, a 422 or 404 wont necessarily have the same business logic explaination.